test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -11132,8 +11150,12 @@ def start(self, sched_home=None, args=None, launcher=None):
             if sched_home is not None:
                 cmd += ['-d', sched_home]
             try:
-                ret = self.du.run_cmd(self.hostname, cmd, sudo=True,
-                                      logerr=False, level=logging.INFOCLI)
+                if self.user.name == 'root':
+                    ret = self.du.run_cmd(self.hostname, cmd, sudo=True,
+                                          logerr=False, level=logging.INFOCLI)
+                else:
+                    ret = self.du.run_cmd(self.hostname, cmd, runas=self.user,
+                                          logerr=False, level=logging.INFOCLI)
```

Comment:
Can't you combine these both and say runas=self.user.name?

Reply:
if we `runas=root`, it runs the command as `sudo -u root qstat` instead of `sudo qstat`. I think our test harness handles these differently

Reply:
Could you give it a try?  It'd clean up the code.